agent/consul/stream/health_endpoint.proto

```diff
@@ -20,6 +20,7 @@ message Event {
     string key = 2;
     uint64 index = 3;
     ServiceHealthUpdate serviceHealth = 4;
+    bool endOfSnapshot = 5;
```

Comment:
As noted above, maybe we should just make this a "snapshot" bool and level trigger rather than edge trigger?

I realise that it becomes more bits on the wire if every snapshot event or every non-snapshot even needs it to be set - few more bytes per message but that seems OK to me.

Alternatively (don't feel strongly) when we make Event type polymorphic, "EndOfSnapshot" should just become one of the types - I thought this was intentionally piggybacking on an existing event message but it's not in current usage then it saves having this single-purpose flag here.

Reply:
The only issue with this is that the client needs to know when the snapshot has ended before receiving any other messages (so that it can send the state to the watcher). With a snapshot = bool field there's no way to know when the snapshot is done being sent and it relies on getting another message that's an actual update before it can return something.